app/services/files/create_service.rb

```diff
@@ -9,9 +9,14 @@ def commit
     def validate
       super
 
-      file_name = File.basename(@file_path)
+      if @file_path =~ Gitlab::Regex.directory_traversal_regex
```

Comment:
I suppose we could, but that seems like a larger refactor. You wouldn't want to catch some invalid names with the file regex and then some others by rescuing an exception.